### PR TITLE
[DONE] Maintenance items for Lizard-Management - part 1

### PR DESCRIPTION
**Pull request by hoanphungt**
ToDo:

- Upgrade to React-scripts 4.0.3, React 17.0.2 and Typescript 4.1.2
- Remove all "any" annotations used to bypass Typescript checks
- Run NPM audit

### CODE REVIEW

**Comment by hoanphungt** [comment]
@remcogerlich 
Could you review this PR for the maintenance items of the lizard-management-client? It's not done yet but it already covered a large part, including:
- Upgrade to React-scripts 4.0.3, React 17.0.2 and Typescript 4.1.2
- Remove "any" from around 700 of them to around 70, most of them are from the table components
- Refactor some .js/.jsx to .ts/.tsx, most importantly the action.js and reducers.js

File: src/data_management/rasters/DeleteRasterSourceNotAllowed.tsx [hunk 1]
```diff
@@ -20,32 +21,32 @@ const DeleteRasterSourceNotAllowed: React.FC<MyProps> = (props) => {
   // const fetchLayers = (uuids: string[],) => {
   //   const baseUrl = "/api/v4/rasters/";
   //   const fetches = uuids.map (uuid => {
-  //     return (fetch(baseUrl + uuid + "/").then((res:any)=>res.json()));
+  //     return (fetch(baseUrl + uuid + "/").then((res)=>res.json()));
   //   });
   //   return Promise.all(fetches)
   // }
```
Comment (remcogerlich) [inline src/data_management/rasters/DeleteRasterSourceNotAllowed.tsx:28]: While we're doing this, could we also remove commented out code?

File: src/reducers.ts [hunk 2]
```diff
@@ -24,49 +23,74 @@ import {
...
+): OrganisationState {
+  // I think this part is outdated --> Need to confirm
   // if there is already a selected organisation then this organisation must not have the unique_id field.
   // if it has the unique_id field then empty the selected organisation
-  if (state.selected && state.selected.unique_id !== undefined) {
-    state.selected = null;
-  }
+  // if (state.selected && state.selected.unique_id !== undefined) {
+  //   state.selected = null;
+  // }
```
Comment (remcogerlich) [inline src/reducers.ts:93]: Think this needs to be resolved also, if we leave it like this it'll just stay like that for a long time

File: src/reducers.ts [hunk 3]
```diff
@@ -165,7 +212,9 @@ function rasterSourceUUID(state = null, action) {
   };
 };
 
-function location(state = null, action) {
+type LocationState = any | null;
```
Comment (remcogerlich) [inline src/reducers.ts:215]: Was this any forgotten?

**Review by remcogerlich — APPROVED** [review]
Just a few very small remarks, it looks great! Very long PR :-)

If you haven't done it yet, could you also run `npx prettier --write --print-width 100 src/` on it? I have started doing that to give everything the same indentation methods etc. And this already touches a lot of files so why not do it here.

File: src/data_management/rasters/DeleteRasterSourceNotAllowed.tsx (hunk 1 above)
Comment (hoanphungt) [reply src/data_management/rasters/DeleteRasterSourceNotAllowed.tsx]: Removed

File: src/reducers.ts (hunk 2 above)
Comment (hoanphungt) [reply src/reducers.ts]: Removed commented out code

**Pull request merged**